### PR TITLE
Change reference spec for the kubernetes events input binding. 

### PR DESCRIPTION
**Pull request by mukundansundar**
Add Role ,RoleBinding doc.

_Please reference the issue this PR will close: #750

### CODE REVIEW

File: reference/specs/bindings/kubernetes.md [hunk 1]
```diff
@@ -10,7 +10,62 @@ spec:
...
   - name: namespace
-    value: default
+    value: <NAMESPACE>
+  - name: resyncPeriodInSec
+    vale: "<seconds>"
 ```
 
-- `namespace` is the Kubernetes namespace to read events from. Default is `default`.
+- `namespace` (required) is the Kubernetes namespace to read events from.
+- `resyncPeriodInSec` (option, default `10`) the period of time to refresh event list from Kubernetes API server.
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:19]: change `option` to `optional`

File: reference/specs/bindings/kubernetes.md [hunk 2]
```diff
@@ -10,7 +10,62 @@ spec:
...
+
+Output received from the binding is of format `bindings.ReadResponse` with the `Data` field populated with the following structure:
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:21]: Put a JSON example here, not a Go example.

File: reference/specs/bindings/kubernetes.md [hunk 3]
```diff
@@ -10,7 +10,62 @@ spec:
...
+Three different event types are available: 
+- Add : Only the NewVal field is populated, OldVal field is an empty `v1.Event`, Event is `add`
+- Delete : Only the OldVal field is populated, NewVal field is an empty `v1.Event`, Event is `delete`
+- Update : Both the OldVal and NewVal fields are populated,  Event is `update`
+
+## Required permisiions
+
+For consuming `events` from Kubernetes, permissions need to be assigned to a User/Group/ServiceAccount using [RBAC Auth] mechanism of Kubernetes.
+
+### For Role
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:39]: Change `For Role` to `Role`

File: reference/specs/bindings/kubernetes.md [hunk 4]
```diff
@@ -10,7 +10,62 @@ spec:
...
+metadata:
+  namespace: <NAMESPACE>
+  name: <ROLENAME>
+rules:
+- apiGroups: [""]
+  resources: ["events"]
+  verbs: ["get", "watch", "list"]
+```
+
+### For RoleBinding
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:55]: `For RoleBinding` to `RoleBinding`

File: reference/specs/bindings/kubernetes.md [hunk 5]
```diff
@@ -10,7 +10,62 @@ spec:
...
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: <BINDINGNAME>
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:61]: this can be any name

File: reference/specs/bindings/kubernetes.md [hunk 6]
```diff
@@ -10,7 +10,85 @@ spec:
...
+- Update : Both the `oldVal` and `newVal` fields are populated,  `event` is `update`
+
+## Required permisiions
+
+For consuming `events` from Kubernetes, permissions need to be assigned to a User/Group/ServiceAccount using [RBAC Auth] mechanism of Kubernetes.
+
+### Role
+
+One of the rules need to be of the form as below to give permissions to `get, watch` and `list` `events`. API Groups can be as restritive as needed.
+
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:65]: Typo, restrictive

File: reference/specs/bindings/kubernetes.md [hunk 7]
```diff
@@ -10,7 +10,85 @@ spec:
...
+- Delete : Only the `oldVal` field is populated, `newVal` field is an empty `v1.Event`, `event` is `delete`
+- Update : Both the `oldVal` and `newVal` fields are populated,  `event` is `update`
+
+## Required permisiions
+
+For consuming `events` from Kubernetes, permissions need to be assigned to a User/Group/ServiceAccount using [RBAC Auth] mechanism of Kubernetes.
+
+### Role
+
+One of the rules need to be of the form as below to give permissions to `get, watch` and `list` `events`. API Groups can be as restritive as needed.
```
Comment (yaron2) [inline reference/specs/bindings/kubernetes.md:64]: Typo, rules should be roles.

Comment (mukundansundar) [reply reference/specs/bindings/kubernetes.md:64]: it is rules ... see the yaml

**Pull request merged**